lldb/source/Plugins/DynamicLoader/FreeBSD-Kernel/DynamicLoaderFreeBSDKernel.cpp

```diff
@@ -200,8 +201,19 @@ lldb_private::UUID DynamicLoaderFreeBSDKernel::CheckForKernelImageAtAddress(
   if (header.e_type != llvm::ELF::ET_EXEC)
     return UUID();
 
-  ModuleSP memory_module_sp =
-      process->ReadModuleFromMemory(FileSpec("temp_freebsd_kernel"), addr);
+  ArchSpec kernel_arch(llvm::ELF::convertEMachineToArchName(header.e_machine));
+
+  ModuleSP memory_module_sp = process->ReadModuleFromMemory(
+      FileSpec("temp_freebsd_kernel"), addr, header.e_shoff);
+  if (header.e_ident[llvm::ELF::EI_CLASS] == llvm::ELF::ELFCLASS64) {
```

Comment:
Does this not happen for 32 bit kernels?

Reply:
```c++
ModuleSP memory_module_sp = process->ReadModuleFromMemory(
      FileSpec("temp_freebsd_kernel"), addr, header.e_shoff);
```

This code segment is already handle the 32bit case as the original header structure is the 32bit ELF Ehdr. I just check if the ELF is 64bit version and replace it with the correct 64bit ELF Ehdr.

Reply:
Ok, so we're relying on the first few fields of the header being the same for 32 and 64 bit. Which is fine.

However I would like to see it as much as possible written as:
```
if 64 bit:
    64 bit stuff
else:
   32 bit stuff
```
Even if it means repeating a couple of things, because the program flow here is getting confusing.